bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -1329,7 +1326,7 @@ bool BCStateTran::onMessage(const CheckpointSummaryMsg *m, uint32_t msgLen, uint
     }
   }
 
-  metrics_.preferred_replicas_.Get().Set(sourceSelector_.preferredReplicasToString());
+  sourceSelector_.metrics_.preferred_replicas_.Get().Set(sourceSelector_.preferredReplicasToString());
```

Comment:
This comment is valid for the rest of the code (general comment).
You decided to handle here more than the task required - essentially you are doing here refactoring (I'm not saying it's bad, just making things clear).
I think that no SS metrics should be handled in ST, since SS is a plain old type struct but a class,I think its metrics should be protected and you should initialize them and control in its own ctor and class logic.
Wherever internal logic doesn't make sense, you might need a new API.
Also, you will probably need some stringify or getter function to print the metrics from ST whenever needed.
If you think it is too much to do, you can revert the refactoring work and leave only what was defined in the task.
Also, I would split the commits, this is 2 different subjects - the refactoring and adding of new metrics for the replacement type.